newIDE/app/src/EventsSheet/ParameterFields/ObjectField.js

```diff
@@ -76,3 +76,6 @@ export default class ObjectField extends React.Component<
     );
   }
 }
+export const renderObjectWithThumbnail = (value: string,getObjectThumbnail) => {
+  return (<span title={value}><img style={{maxWidth:48}} width={'auto'} height={24} src={getObjectThumbnail? getObjectThumbnail(value):''}></img>{value}</span>);
```

Comment:
All the thumbnail/icons should be square and with a small size, otherwise you're getting huge image like for bullets (that are wide) or player (that is tall).
Use maxWidth and maxHeight. Use a size like 24x24, 20x20 or even less if needed. When you've decided on a size, use it for all icons. 

Also add margins on the right like for variables.

Reply:
The thumbnails are a bit of a special case , since their aspect ratio might not be 1x1 like the icons.
In that case my idea is to make their height the same as the icons, but their width can automatically adapt, so as not to stretch/squish them- that is of course with a maxWidth limitation to stop any super wide sprites from taking too much horizontal space - if too wide-only then squash.

For the icons there seems to be controversy with some users insisting to stick with 16x16 and others saying that we need bigger icons (20-24). I don't mind either way,but have to come to a decision for the size. I personally like 24 :)

Reply:
For consistency, we have no choices but using the size of the action/conditions icons on the left (can't imagine a good design with icons that have different sizes on the same lines). Size is 16*16 with 2 pixels padding on the left and right.